modules/backend/widgets/form/partials/_field_switch.htm

```diff
@@ -1,3 +1,8 @@
+<?php
+    if($field->readOnly){
+        $this->previewMode = true;
```

Comment:
I just realized, this will cause an unintentional bug when setting readOnly on a single switch field, it will cause the entire form after the switch field to be in previewMode. To fix it, use this code at the beginning of the file to set a local flag instead of relying on the Form object's global `previewMode` flag:
```php
<?php
    $previewMode = false;
    if ($this->previewMode || $field->readOnly) { 
        $previewMode = true;
    }
?>
```
Then replace all references to the global flag `$this->previewMode` in this partial to just be the local flag `$previewMode`